typescript/sdk/src/token/EvmERC20WarpRouteReader.ts

```diff
@@ -1,33 +1,42 @@
-import { ethers, providers } from 'ethers';
+import { BigNumber, constants, providers } from 'ethers';
 
 import {
-  ERC20__factory,
+  HypERC20CollateralVaultDeposit__factory,
   HypERC20Collateral__factory,
-  MailboxClient__factory,
+  HypERC20__factory,
 } from '@hyperlane-xyz/core';
-import { Address, eqAddress } from '@hyperlane-xyz/utils';
+import { HookConfig, TokenRouterConfig, TokenType } from '@hyperlane-xyz/sdk';
+import { Address, eqAddress, rootLogger } from '@hyperlane-xyz/utils';
 
 import { DEFAULT_CONTRACT_READ_CONCURRENCY } from '../consts/concurrency.js';
-import { EvmHookReader } from '../hook/EvmHookReader.js';
-import { EvmIsmReader } from '../ism/EvmIsmReader.js';
+import { DerivedHookConfig, EvmHookReader } from '../hook/EvmHookReader.js';
+import { DerivedIsmConfig, EvmIsmReader } from '../ism/EvmIsmReader.js';
 import { MultiProvider } from '../providers/MultiProvider.js';
-import { MailboxClientConfig } from '../router/types.js';
-import { ChainName } from '../types.js';
+import { ChainNameOrId } from '../types.js';
 
-import { TokenType } from './config.js';
-import { TokenRouterConfig } from './schemas.js';
 import { TokenMetadata } from './types.js';
 
-const { AddressZero } = ethers.constants;
+type WarpRouteBaseMetadata = Record<'mailbox' | 'owner', string> & {
+  hook?: HookConfig;
+  interchainSecurityModule?: DerivedIsmConfig;
+};
```

Comment:
I am not a fan of "metadata" as a name
further I think this is just `MailboxClient`